app/components/channel_drawer/channels_list/channel_unread_item/index.js

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannelId, getChannel, getMyChannelMember} from 'mattermost-redux/selectors/entities/channels';
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import ChannelUnreadItem from './channel_unread_item';
+
+function mapStateToProps(state, ownProps) {
+    const channel = getChannel(state, ownProps.channelId);
+    const member = getMyChannelMember(state, ownProps.channelId);
+
+    return {
+        currentChannelId: getCurrentChannelId(state),
+        displayName: channel.display_name,
+        hasUnread: channel.total_msg_count - member.msg_count > 0,
+        mentions: member.mention_count,
+        status: channel.status,
+        theme: getTheme(state),
+        type: channel.type
+    };
+}
+
+function areStatesEqual(next, prev) {
```

Comment:
Same comment here as with the other one. All of these selectors are efficient, so I don't think this is needed

Reply:
can remove both, although we should try and call the selectors no matter how efficient they are only when the state changes in places that we care about

Reply:
I think it's roughly equivalent no matter how we do it, so we shouldn't need to do it twice. Either the selectors are checking if the state changed, or areStatesEqual is doing it, so we probably don't need to check it twice